cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -165,4 +180,83 @@ public void centers(HttpServiceRequest request, HttpServiceResponder responder,
       }
     }
   }
+
+  /**
+   * A Service which serves the number of requests made by unique ip address from a {@link TimePartitionedFileSet}
+   */
+  public static final class RequestCounterHandler extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+    static final String REQUEST_COUNTER_PARTITIONS_PATH = "reqcount";
+    static final String REQUEST_FILE_CONTENT_PATH = "reqfile";
+    static final String REQUEST_FILE_PATH_HANDLER_KEY = "time";
+
+    @UseDataSet(REQ_COUNT_STORE)
+    private TimePartitionedFileSet reqCountStore;
+
+    /**
+     * Handler which lists all the different time partitions available in the {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet}
+     */
+    @Path(REQUEST_COUNTER_PARTITIONS_PATH)
+    @GET
+    public void getRequestFilesetPartitions(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      // get all the existing paritions
+      Set<TimePartitionDetail> partitionsByTime = reqCountStore.getPartitionsByTime(0, Long.MAX_VALUE);
+
+      SortedSet<String> formattedTimes = new TreeSet<>();
+      for (TimePartitionDetail timePartitionDetail : partitionsByTime) {
+        String partitionTime = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT).format(
```

Comment:
It's pretty expensive to create the DataFormat inside a loop, while it is not changing. Since CDAP guarantees there would be no concurrent calls made to the same handler instance, you can make the DateFormat a static constant.